### PR TITLE
feat: override metric format

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #7339

### Description:

Note: There is a caveat with MIN/MAX metrics which we consider as numeric metrics across the app when the dimension is a date/timestamp. I'm not addressing that problem in this PR.

Changes:
- allow to edit numeric metric format in the results table
- allow to reset format
- save new format
- new format is applied to results table, chart, csv downloads, scheduled deliveries

## Demo:

Results table
Screenshot 2024-12-23 at 09 46 57
Screenshot 2024-12-23 at 09 47 04

Chart viz
Screenshot 2024-12-23 at 09 46 08
Screenshot 2024-12-23 at 09 46 41

Export
Screenshot 2024-12-23 at 09 45 21

Dashboard tile
Screenshot 2024-12-23 at 10 31 37

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/CsvService/CsvService.ts [hunk 1]
```diff
@@ -1167,7 +1167,7 @@ This method can be memory intensive
                 organization_uuid: user.organizationUuid,
             };
 
-            const { rows } = await this.projectService.runMetricQuery({
```
Comment (ZeRego) [inline packages/backend/src/services/CsvService/CsvService.ts:1170]: Opmitize/fix CSV service to use fields from the query method.

File: packages/common/src/types/metricQuery.ts [hunk 2]
```diff
@@ -46,12 +47,13 @@ export const isAdditionalMetric = (value: any): value is AdditionalMetric =>
 
 export const hasFormatOptions = (
     value: any,
-): value is AdditionalMetric & { formatOptions: CustomFormat } =>
-    !!value.formatOptions;
+): value is { formatOptions: CustomFormat } => !!value.formatOptions;
```
Comment (ZeRego) [inline packages/common/src/types/metricQuery.ts:50]: Fix util function as this property is in both Metric and AdditionalMetric.

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 3]
```diff
@@ -2074,6 +2092,11 @@ export class ProjectService extends BaseService {
                             ...(explore.type === ExploreType.VIRTUAL
                                 ? { virtualViewId: explore.name }
                                 : {}),
+                            metricOverridesCount: Object.keys(
```
Comment (magnew) [inline packages/backend/src/services/ProjectService/ProjectService.ts:2095]: Do you store a count just to avoid computing it every time?

**Review by magnew — APPROVED** [review]
Very cool!
✅  edit numeric metric
✅ reset works
✅ save works
✅ table, csv, chart and scheduled deliveries all work

Approved with a couple things to fix if possible:

- The total row for a metric that previously had formatting keeps the old formatting
Screenshot 2024-12-23 at 15 24 42

- The number of places box is a little to small to include the word 'places'
Screenshot 2024-12-23 at 15 34 34

**Pull request merged**